lib/chef/provider/package/snap.rb

```diff
@@ -0,0 +1,318 @@
+#
+# Author:: S.Cavallo (<smcavallo@hotmail.com>)
+# Copyright:: Copyright 2016-2018, Chef Software Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require "chef/provider/package"
+require "chef/resource/snap_package"
+require "chef/mixin/shell_out"
+require 'socket'
+require 'json'
+
+class Chef
+  class Provider
+    class Package
+      class Snap < Chef::Provider::Package
+        extend Chef::Mixin::Which
+        extend Chef::Mixin::ShellOut
+
+        allow_nils
+        use_multipackage_api
+
+        # Todo: support non-debian platforms
+        provides :package, platform_family: "debian"
```

Comment:
oh yeah you can't do that.

you're wiring this up as the default package provider for everyone on debian here.

as it looks like snap do not wholesale replace apt_package on debian, you probably can't do this and need to delete this line.

Reply:
(probably the source of most of the unit test issues as well)